evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -70,6 +74,8 @@ def transform(self, X, y=None):
         # Convert None to np.nan, since None cannot be properly handled
         X = X.fillna(value=np.nan)
 
+        if (X.dtypes == bool).all():
```

Comment:
Since this is kinda confusing, perhaps add a comment saying "bool dtype doesn't support nans, and sklearn errors if all cols are bool, so return early"